--- usercp_register.php.orig	2007-06-25 15:20:20.953211000 -0700
+++ usercp_register.php	2007-06-25 15:18:57.237525000 -0700
@@ -40,6 +40,11 @@
 	exit;
 }
 
+require_once('recaptchalib.php');
+$recaptcha_public_key = ''; # You need to put your public and private key here
+$recaptcha_private_key = '';
+$recaptcha_code = null;
+
 $unhtml_specialchars_match = array('#&gt;#', '#&lt;#', '#&quot;#', '#&amp;#');
 $unhtml_specialchars_replace = array('>', '<', '"', '&');
 
@@ -281,53 +278,14 @@
 
 	if ($board_config['enable_confirm'] && $mode == 'register')
 	{
-		if (empty($HTTP_POST_VARS['confirm_id']))
-		{
+	
+		$recaptcha_response =  recaptcha_check_answer ($recaptcha_private_key, $client_ip, $HTTP_POST_VARS['recaptcha_challenge_field'], $HTTP_POST_VARS['recaptcha_response_field']); 	
+		if (!$recaptcha_response->is_valid) {
 			$error = TRUE;
+			$recaptcha_code = recaptcha_get_html ($recaptcha_public_key, $recaptcha_response->error);
 			$error_msg .= ( ( isset($error_msg) ) ? '<br />' : '' ) . $lang['Confirm_code_wrong'];
 		}
-		else
-		{
-			$confirm_id = htmlspecialchars($HTTP_POST_VARS['confirm_id']);
-			if (!preg_match('/^[A-Za-z0-9]+$/', $confirm_id))
-			{
-				$confirm_id = '';
-			}
-			
-			$sql = 'SELECT code 
-				FROM ' . CONFIRM_TABLE . " 
-				WHERE confirm_id = '$confirm_id' 
-					AND session_id = '" . $userdata['session_id'] . "'";
-			if (!($result = $db->sql_query($sql)))
-			{
-				message_die(GENERAL_ERROR, 'Could not obtain confirmation code', '', __LINE__, __FILE__, $sql);
-			}
 
-			if ($row = $db->sql_fetchrow($result))
-			{
-				if ($row['code'] != $confirm_code)
-				{
-					$error = TRUE;
-					$error_msg .= ( ( isset($error_msg) ) ? '<br />' : '' ) . $lang['Confirm_code_wrong'];
-				}
-				else
-				{
-					$sql = 'DELETE FROM ' . CONFIRM_TABLE . " 
-						WHERE confirm_id = '$confirm_id' 
-							AND session_id = '" . $userdata['session_id'] . "'";
-					if (!$db->sql_query($sql))
-					{
-						message_die(GENERAL_ERROR, 'Could not delete confirmation code', '', __LINE__, __FILE__, $sql);
-					}
-				}
-			}
-			else
-			{		
-				$error = TRUE;
-				$error_msg .= ( ( isset($error_msg) ) ? '<br />' : '' ) . $lang['Confirm_code_wrong'];
-			}
-			$db->sql_freeresult($result);
-		}
 	}
 
 	$passwd_sql = '';
@@ -952,70 +909,11 @@
 
 
 	// Visual Confirmation
-	$confirm_image = '';
 	if (!empty($board_config['enable_confirm']) && $mode == 'register')
 	{
-		$sql = 'SELECT session_id 
-			FROM ' . SESSIONS_TABLE; 
-		if (!($result = $db->sql_query($sql)))
-		{
-			message_die(GENERAL_ERROR, 'Could not select session data', '', __LINE__, __FILE__, $sql);
-		}
-
-		if ($row = $db->sql_fetchrow($result))
-		{
-			$confirm_sql = '';
-			do
-			{
-				$confirm_sql .= (($confirm_sql != '') ? ', ' : '') . "'" . $row['session_id'] . "'";
-			}
-			while ($row = $db->sql_fetchrow($result));
+		if (!$recaptcha_code)
+		$recaptcha_code = recaptcha_get_html ($recaptcha_public_key, null);	
 		
-			$sql = 'DELETE FROM ' .  CONFIRM_TABLE . " 
-				WHERE session_id NOT IN ($confirm_sql)";
-			if (!$db->sql_query($sql))
-			{
-				message_die(GENERAL_ERROR, 'Could not delete stale confirm data', '', __LINE__, __FILE__, $sql);
-			}
-		}
-		$db->sql_freeresult($result);
-
-		$sql = 'SELECT COUNT(session_id) AS attempts 
-			FROM ' . CONFIRM_TABLE . " 
-			WHERE session_id = '" . $userdata['session_id'] . "'";
-		if (!($result = $db->sql_query($sql)))
-		{
-			message_die(GENERAL_ERROR, 'Could not obtain confirm code count', '', __LINE__, __FILE__, $sql);
-		}
-
-		if ($row = $db->sql_fetchrow($result))
-		{
-			if ($row['attempts'] > 3)
-			{
-				message_die(GENERAL_MESSAGE, $lang['Too_many_registers']);
-			}
-		}
-		$db->sql_freeresult($result);
-		
-		// Generate the required confirmation code
-		// NB 0 (zero) could get confused with O (the letter) so we make change it
-		$code = dss_rand();
-		$code = substr(str_replace('0', 'Z', strtoupper(base_convert($code, 16, 35))), 2, 6);
-
-		$confirm_id = md5(uniqid($user_ip));
-
-		$sql = 'INSERT INTO ' . CONFIRM_TABLE . " (confirm_id, session_id, code) 
-			VALUES ('$confirm_id', '". $userdata['session_id'] . "', '$code')";
-		if (!$db->sql_query($sql))
-		{
-			message_die(GENERAL_ERROR, 'Could not insert new confirm code information', '', __LINE__, __FILE__, $sql);
-		}
-
-		unset($code);
-		
-		$confirm_image = '<img src="' . append_sid("profile.$phpEx?mode=confirm&amp;id=$confirm_id") . '" alt="" title="" />';
-		$s_hidden_fields .= '<input type="hidden" name="confirm_id" value="' . $confirm_id . '" />';
-
 		$template->assign_block_vars('switch_confirm', array());
 	}
 
@@ -1033,7 +931,7 @@
 		'NEW_PASSWORD' => isset($new_password) ? $new_password : '',
 		'PASSWORD_CONFIRM' => isset($password_confirm) ? $password_confirm : '',
 		'EMAIL' => isset($email) ? $email : '',
-		'CONFIRM_IMG' => $confirm_image, 
+		'RECAPTCHA_CODE' => $recaptcha_code, 
 		'YIM' => $yim,
 		'ICQ' => $icq,
 		'MSN' => $msn,
